core/admin/mailu/models.py

```diff
@@ -259,16 +259,16 @@ def dns_autoconfig(self):
         """ return list of auto configuration records (RFC6186) """
         hostname = app.config['HOSTNAME']
         protocols = [
-            ('imap', 143, 20),
-            ('pop3', 110, 20),
-            ('submission', 587, 20),
+            ('imap', 143, 20 if '143' in app.config['PORTS'].split(',') else 0),
```

Comment:
I think you should normalize ports before (line 161) like this
`ports = {port.strip() for port in app.config['PORTS'].split(",")}`
And use `if '143' in ports else 0` here and below.

We really should start normalizing the options at a central place (in the Config class).

Reply:
done in cdbfa4ee01e87ed02e9e0e5e129a6b19c8901f20